src/service/publish-service.js

```diff
@@ -67,7 +67,7 @@ class PublishService extends OperationService {
             }
         } else if (
             numberOfFoundNodes === numberOfResponses ||
-            numberOfNodesInBatch === numberOfResponses
+            numberOfResponses % numberOfNodesInBatch === 0
```

Comment:
I think it could happen that we mark it as failed after marking it as completed.

1. Not enough completed responses and node retries until last batch
2. First responses are ACKs, node marks publish as completed
3. When last response is processed, leftover nodes === 0 and node marks operation as failed